### PR TITLE
Fix building of examples

### PR DESCRIPTION
**Pull request by atiselsts**
Solves problems found with the `build-all.sh` script.

### CODE REVIEW

**Comment by simonduq** [comment]
Update:
```
Number of examples skipped or built successfully: 866
Number of examples that failed to build: 52
Failed examples: ;
../../examples/hello-world for cc2538dk (default);
../../examples/http-socket for jn516x (default);
../../examples/slip-radio for nrf52dk (default);
../../examples/coap for nrf52dk (default);
../../examples/http-socket for nrf52dk (default);
../../examples/rpl-border-router for nrf52dk (default);
../../examples/platform-specific/nrf52dk/coap-demo/coap-server for nrf52dk (default);
../../examples/rpl-udp for nrf52dk (default);
../../examples/multicast for nrf52dk (default);
../../examples/sensniff for openmote-cc2538 (default);
../../examples/ip64-router for zoul (remote-reva);
../../examples/ip64-router for zoul (remote-revb);
../../examples/ip64-router for zoul (firefly-reva);
../../examples/ip64-router for zoul (firefly);
../../examples/coap for zoul (orion);
../../examples/6tisch/sixtop for zoul (orion);
../../examples/6tisch/6p-packet for zoul (orion);
../../examples/6tisch/etsi-plugtest-2017 for zoul (orion);
../../examples/6tisch/simple-node for zoul (orion);
../../examples/http-socket for zoul (orion);
../../examples/rpl-border-router for zoul (orion);
../../examples/platform-specific/cc2538-common/mqtt-demo for zoul (orion);
../../examples/platform-specific/zoul/rev-b for zoul (remote-reva);
../../examples/platform-specific/zoul/rev-b for zoul (remote-revb);
../../examples/platform-specific/zoul/rev-b for zoul (firefly-reva);
../../examples/platform-specific/zoul/rev-b for zoul (firefly);
../../examples/platform-specific/zoul/rev-b for zoul (orion);
../../examples/platform-specific/zoul/rtcc for zoul (remote-reva);
../../examples/platform-specific/zoul/rtcc for zoul (remote-revb);
../../examples/platform-specific/zoul/rtcc for zoul (firefly-reva);
../../examples/platform-specific/zoul/rtcc for zoul (firefly);
../../examples/platform-specific/zoul/rtcc for zoul (orion);
../../examples/platform-specific/zoul/at-test for zoul (remote-reva);
../../examples/platform-specific/zoul/at-test for zoul (remote-revb);
../../examples/platform-specific/zoul/at-test for zoul (firefly-reva);
../../examples/platform-specific/zoul/at-test for zoul (firefly);
../../examples/platform-specific/zoul/at-test for zoul (orion);
../../examples/platform-specific/zoul/orion/ip64-router for zoul (remote-reva);
../../examples/platform-specific/zoul/orion/ip64-router for zoul (remote-revb);
../../examples/platform-specific/zoul/orion/ip64-router for zoul (firefly-reva);
../../examples/platform-specific/zoul/orion/ip64-router for zoul (firefly);
../../examples/platform-specific/zoul/orion/client for zoul (remote-reva);
../../examples/platform-specific/zoul/orion/client for zoul (remote-revb);
../../examples/platform-specific/zoul/orion/client for zoul (firefly-reva);
../../examples/platform-specific/zoul/orion/client for zoul (firefly);
../../examples/platform-specific/zoul/orion/client for zoul (orion);
../../examples/nullnet for zoul (orion);
../../examples/multicast for zoul (remote-reva);
../../examples/multicast for zoul (remote-revb);
../../examples/multicast for zoul (firefly-reva);
../../examples/multicast for zoul (firefly);
../../examples/multicast for zoul (orion)
```

**Comment by g-oikonomou** [comment]
I just got round to looking at this

> ../../examples/hello-world for cc2538dk (default);
> ../../examples/sensniff for openmote-cc2538 (default);
Build locally

> ../../examples/multicast for zoul (remote-reva);
> ../../examples/multicast for zoul (remote-revb);
> ../../examples/multicast for zoul (firefly-reva);
> ../../examples/multicast for zoul (firefly);
> ../../examples/multicast for zoul (orion)

Out of RAM problems here. Those will build if we change to `LPM_CONF_MAX_PM 1` (gives us +16KB of RAM)

> ../../examples/platform-specific/zoul/rev-b for zoul (remote-reva);
> ../../examples/platform-specific/zoul/rev-b for zoul (remote-revb);
> ../../examples/platform-specific/zoul/rev-b for zoul (firefly-reva);
> ../../examples/platform-specific/zoul/rev-b for zoul (firefly);
> ../../examples/platform-specific/zoul/rev-b for zoul (orion);
> ../../examples/platform-specific/zoul/rtcc for zoul (remote-reva);
> ../../examples/platform-specific/zoul/rtcc for zoul (remote-revb);
> ../../examples/platform-specific/zoul/rtcc for zoul (firefly-reva);
> ../../examples/platform-specific/zoul/rtcc for zoul (firefly);
> ../../examples/platform-specific/zoul/rtcc for zoul (orion);
Fixed in #327

**Comment by simonduq** [comment]
OK I also have some pending fixes at https://github.com/atiselsts/contiki-ng/pull/2

**Comment by simonduq** [comment]
BTW the ones you mention build locally also build in Docker. But somehow failed from `build-all.sh`, no clue why

**Comment by g-oikonomou** [comment]
Feel free to cherry-pick as required from #327, I'll close it now.

**Comment by atiselsts** [comment]
Not sure why the build fails...

**Comment by g-oikonomou** [comment]
I was wondering about that too. Please tell me it's not because there are two spaces before the backslash in the Makefile...

**Comment by simonduq** [comment]
I think it's because the path should be `platform-specific/zoul/rev-b`, not `platform-specific/zoul/rev-b/test-power-mgmt`

**Comment by simonduq** [comment]
(same goes with all the subsequent lines)

**Comment by g-oikonomou** [comment]
Ugh of course!

Also means that we only need a single line (per board) for examples in the top-level zoul dir.

**Comment by atiselsts** [comment]
Could you submit a fix for this, George?

**Comment by g-oikonomou** [comment]
I pushed a commit to #327

**Comment by atiselsts** [comment]
Ready now. Let me know if there are any more changes to include, otherwise can we merge this?

File: examples/platform-specific/zoul/orion/client/Makefile.target [hunk 1]
```diff
@@ -1 +1,2 @@
 TARGET = zoul
+BOARD = orion
```
Comment (g-oikonomou) [inline examples/platform-specific/zoul/orion/client/Makefile.target]: @simonduq made a comment about this one not being orion-specific (if that's the case then I do find the path misleading). I am not sure myself, I did not look into it.

Comment (simonduq) [reply examples/platform-specific/zoul/orion/client/Makefile.target]: Yes it looks like `ip64-router` is meant for the Orion, but `client` is just an example HTTP client used to try out an IP64 Orion router.

Comment (g-oikonomou) [reply examples/platform-specific/zoul/orion/client/Makefile.target]: Let's then just undo my change to the Makefile.target and merge.

Comment (atiselsts) [reply examples/platform-specific/zoul/orion/client/Makefile.target]: Removed.

**Pull request merged**